ecl/hql/hqlmeta.cpp

```diff
@@ -1572,10 +1572,17 @@ IHqlExpression * ensureSorted(IHqlExpression * dataset, IHqlExpression * order,
                 return subsorted.getClear();
         }
     }
-
-    IHqlExpression * attr1 = isLocal ? createLocalAttribute() : (isGrouped(dataset) && ignoreGrouping) ? createAttribute(globalAtom) : NULL;
-    IHqlExpression * attr2 = requestSpilling ? createAttribute(spillAtom) : NULL;
-    return createDatasetF(no_sort, LINK(dataset), LINK(order), createComma(attr1, attr2), NULL);
+    HqlExprArray args;
+    args.append(OLINK(*dataset));
+    args.append(OLINK(*order));
+    if (isLocal)
+        args.append(*createLocalAttribute());
+    else if (isGrouped(dataset) && ignoreGrouping)
+        args.append(*createAttribute(globalAtom));
+    if (requestSpilling)
+        args.append(*createAttribute(spillAtom));
+    unwindHintAttrs(args, parentExpr);
```

Comment:
I might have made this conditional on parentExpr != NULL, but happy to merge as is.